### PR TITLE
[DEV APPROVED] Update to rails 4.2 and gems

### PR DESCRIPTION
**Pull request by willhall88**
The main goal of this work was to update the project to run on rails 4.2. 
Due to the lack of maintenance of this repo it also needed updating of FactoryBot.

There was a LOT of noise when running the tests. The work in this PR addresses most of that as well as ensuring that all tests continue to pass on 4.2.
There is still a lot of noise coming from paperclip gem. However I am reluctant to touch/update/play with this any further at this stage for two reasons:
- we're already making a lot of changes to our cms and updating this gem may carry risks that might get missed with all the changes currently happening
- [paperclip gem](https://github.com/thoughtbot/paperclip) is deprecated and should be removed and switched to using Rails ActiveStorage instead. Again doing this work, I'm reluctant to do at this point in time.

### CODE REVIEW

File: app/models/comfy/cms/block.rb [hunk 1]
```diff
@@ -1,59 +1,70 @@
...
     # correctly triggering dirty
-    value = nil               if self.temp_files.present?
-    write_attribute(:content, value)
-    self.content_will_change! if self.temp_files.present?
+    if temp_files.present?
+      write_attribute(:content, nil)
+      content_will_change!
+    else
+      write_attribute(:content, value)
+    end
```
Comment (your) [inline app/models/comfy/cms/block.rb]: Could use a guard clause as well, but I think this is already a huuuge improvement compared to what we had... 🤦‍♂️

File: app/models/comfy/cms/block.rb [hunk 2]
```diff
@@ -1,59 +1,70 @@
...
     end
   end
+
+  def pagefile?
+    tag.instance_of?(ComfortableMexicanSofa::Tag::PageFile)
+  end
+
+  def restrict_to_single_file!
+    self.temp_files = [temp_files.first].compact
+    files.map(&:mark_for_destruction)
```
Comment (your) [inline app/models/comfy/cms/block.rb]:
```ruby
file.each(&:mark_for_destruction)
```

should work fine if you don't need to build a new array with `map`, but just return the original.

Again... huuuge improvement 💯

Comment (scruti) [reply app/models/comfy/cms/block.rb]: oh true! Good catch! :+1:

File: app/models/comfy/cms/block.rb (hunk 1 above)
Comment (scruti) [reply app/models/comfy/cms/block.rb]: I personally tend to favour classical `if/else` over guard clauses unless guarding against extreme cases or having multiple levels of conditionals. 
When is a single "do A or B depending on some condition" not falling under any of those cases I prefer the classical if else. But is a totally biased thing :P

**Pull request merged**